src/main/java/com/javapractice/interactors/addTodoInteractor.java

```diff
@@ -0,0 +1,26 @@
+package com.javapractice.interactors;
+
+import org.springframework.stereotype.Service;
+
+import com.javapractice.domain.Request.TodoRequest;
+import com.javapractice.domain.Response.TodoResponse;
+import com.javapractice.gateways.ITodoRepository;
+import com.javapractice.interactors.interfaces.IaddTodo;
+
+@Service
+public class addTodoInteractor implements IaddTodo{
+    private final ITodoRepository todoRepository;
+    
+    public addTodoInteractor(ITodoRepository todoRepository) {
```

Comment:
Notice how the ITodoRepository interface (previous file) is injected into the interactor. This is the boundary between the interactor and the gateway.

Reply:
The implementation of that interface is here. https://github.com/sprintup/javapractice/pull/1/files#r1361116352